### PR TITLE
Support govcloud regions and remove environment hardcoding

### PR DESCRIPTION
**Pull request by dynamike**
This PR allows us to run the tool against GovCloud regions, it also introduces basic tests and runs them in CircleCI.

### CODE REVIEW

**Review by eeeady — APPROVED** [review]

File: main.go [hunk 1]
```diff
@@ -182,20 +173,27 @@ func initFlags(flag *pflag.FlagSet) {
...
+func checkRegion(v *viper.Viper) error {
 	region := v.GetString(flagAWSRegion)
 	if len(region) == 0 {
 		return errors.Wrap(&errInvalidRegion{Region: region}, fmt.Sprintf("%q is invalid", flagAWSRegion))
 	}
+	// PartitionForRegion checks for the first region in a partition
+	// and for regions matching a regex pattern.
+	// So this checks if a region exists in any of the SDK regions, which are
+	// AWS Standard, AWS China, AWS GovCloud, AWS ISO, and AWS ISOB.
+	if _, ok := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); !ok {
```
Comment (kilbergr) [inline main.go:185]: WOW This is GREAT!!!!

File: main.go [hunk 2]
```diff
@@ -242,16 +240,6 @@ func checkConfig(v *viper.Viper) error {
 		if len(environment) == 0 {
 			return errors.New("when status is set to STOPPED then environment must be set")
 		}
-		valid := false
```
Comment (kilbergr) [inline main.go:245]: I totally get why you don't want to be checking against a bunch of strings that were simply names we decided for, but is there any other way to check that we are using the environments we anticipated? What happens when someone passes in an unusable env? Do we catch it elsewhere?

**Review by kilbergr — COMMENTED** [review]
I'm thrilled with the removal of hardcoded values and the partitioning. Are you concerned with not checking envs?

File: main.go (hunk 2 above)
Comment (dynamike) [reply main.go:245]: If you put in a name that doesn't exist you'll get an error say as such
```
panic: error describing log streams: ResourceNotFoundException: The specified log group does not exist.
```

**Review by kilbergr — APPROVED** [review]
🧇

**Pull request merged**